collated/main/kennard123661.md

```diff
@@ -0,0 +1,884 @@
+###### \java\seedu\address\logic\commands\ImportCommand.java
+``` java
+/**
+ * Imports the parcels and tags stored in {@code parcels} into the current instance of Ark.
+ */
+public class ImportCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "import";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds the data of a storage file stored in "
+            + "data/import/ directory into Ark.\n"
+            + "Parameters: FILE (Must be a valid addressbook file stored in xml format) i.e.\n"
+            + "Example: " + COMMAND_WORD + " ark_storage";
+
+    public static final String MESSAGE_SUCCESS_SUMMARY = "Summary: %1$d parcels added and %2$d duplicate "
+            + "parcels not added.\n";
+    public static final String MESSAGE_SUCCESS_BODY = "Parcels added: %3$s\nDuplicate Parcels: %4$s";
+    public static final String MESSAGE_SUCCESS = MESSAGE_SUCCESS_SUMMARY + MESSAGE_SUCCESS_BODY;
+
+    public static final String MESSAGE_FAILURE_DUPLICATE_PARCELS = "All parcels in the imported save file will create "
+            + "duplicate parcels";
+
+    private final List<ReadOnlyParcel> parcels;
+
+    /**
+     * Creates an ImportCommand to add all {@code ReadOnlyParcel}s in {@param parcelList}.
+     */
+    public ImportCommand(List<ReadOnlyParcel> parcelList) {
+        parcels = parcelList;
+    }
+
+    /**
+     * Adds all unique parcels in {@code parcels} to the {@link UniqueParcelList} of the {@link AddressBook}. Ignores
+     * parcels that will create duplicates in the {@link UniqueParcelList}
+     *
+     * @return {@link CommandResult} created by {@link ImportCommand#executeUndoableCommand()}
+     * @throws CommandException if {@code parcels} contain only duplicate parcels.
+     */
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+        requireNonNull(model);
+
+        List<ReadOnlyParcel> uniqueParcels = new ArrayList<>(); // list of unique parcels added to Ark.
+        List<ReadOnlyParcel> duplicateParcels = new ArrayList<>(); // list of duplicate parcels that are not added
+        List<ReadOnlyParcel> storedParcels = model.getAddressBook().getParcelList(); // parcels already stored in Ark
+
+        // check if all parcels are duplicates
+        if (storedParcels.containsAll(parcels)) {
+            throw new CommandException(MESSAGE_FAILURE_DUPLICATE_PARCELS);
+        }
+
+        model.addAllParcels(parcels, uniqueParcels, duplicateParcels);
+
+        String addedParcelsString = getImportFormattedParcelListString(uniqueParcels);
+        String duplicatedParcelsString = getImportFormattedParcelListString(duplicateParcels);
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS, uniqueParcels.size(), duplicateParcels.size(),
+                addedParcelsString, duplicatedParcelsString));
+    }
+
+    /**
+     * @return formatted list of parcels added/not added for ImportCommand execution result.
```

Comment:
The start of the header comment does not follow the convention (i.e. first sentence should be something like `Selects next line …` instead of `Select next line` or `This method is for selecting …`)